connectors/byoc.py

```diff
@@ -554,3 +554,77 @@ async def sync(self, elastic_server, idling, sync_now=False):
         finally:
             self._syncing = False
             self._start_time = None
+
+STUCK_JOBS_THRESHOLD = 60 # 60 seconds
+
+class SyncJobIndex(ESIndex):
+    """
+    Represents Elasticsearch index for sync jobs
+
+    Args:
+        index_name (str): Name of an Elasticsearch index
+        elastic_config (dict): Elasticsearch configuration and credentials
+    """
+    def __init__(self, index_name, elastic_config):
```

Comment:
`JOBS_INDEX` constant doesn't belong to `SyncJobIndex` that's why I didn't use it inside the class. Also, I was thinking about a potential case when you need to use different job indices (e.g. if we introduce another type of job).

Reply:
`JOBS_INDEX` is from the same `byoc` module so it can be easily referenced, like `BYOIndex`:

https://github.com/elastic/connectors-python/blob/8f9be7f7f6d26098e6c854f3232fcc9336903ffb/connectors/byoc.py#L83-L87

BTW, I do think `BYOIndex` should be renamed to `ConnectorIndex`, not scope of this PR.

>Also, I was thinking about a potential case when you need to use different job indices

A new index deserves a new class? From my understanding, any `EsIndex` class represent an ES index.

Reply:
> A new index deserves a new class? From my understanding, any `EsIndex` class represent an ES index.

`EsIndex` holds shared logic between indices like query pagination and yes I think all service-related indices should be presented as objects of their own classes. This will can help a lot during debugging and troubleshooting because engineers will know what they are dealing with.

> BTW, I do think BYOIndex should be renamed to ConnectorIndex, not scope of this PR.
Agree

Reply:
Rename PR: https://github.com/elastic/connectors-python/pull/342

Reply:
>yes I think all service-related indices should be presented as objects of their own classes.

That means there will be an index associated to each derived class of `EsIndex` class, therefore the `index_name` of `SyncJobIndex` should be fixed to `JOBS_INDEX`

Reply:
alright, if it's a blocker I will change this.

Reply:
Let's keep it consistent with `ConnectorIndex`:

https://github.com/elastic/connectors-python/blob/aeb8e5fb15a7bfea0c9db6a885bffa5ed147ba42/connectors/byoc.py#L86

Reply:
What exactly is not consistent with `ConnectorIndex`?

https://github.com/elastic/connectors-python/pull/332/files#diff-b9a5d063adca223525457d0a3c06df668bd814fe46069dda6678b694c277edc9R607

Reply:
Hah, didn't see you have updated it.